jhove-modules/xml-hul/src/main/java/edu/harvard/hul/ois/jhove/module/xml/XmlModuleHandler.java

```diff
@@ -349,19 +350,9 @@ public InputSource resolveEntity(String publicId, String systemId)
 		if (ent == null) {
 			try {
 				URL obj = new URL(systemId);
-				HttpURLConnection conn = (HttpURLConnection) obj
-						.openConnection();
-
-				int status = conn.getResponseCode();
-				if (status == HttpURLConnection.HTTP_MOVED_TEMP
-						|| status == HttpURLConnection.HTTP_MOVED_PERM
-						|| status == HttpURLConnection.HTTP_SEE_OTHER) {
-
-					String newUrl = conn.getHeaderField("Location");
-					conn = (HttpURLConnection) new URL(newUrl).openConnection();
-					ent = new InputSource(conn.getInputStream());
-				}
-
+				ent = new InputSource(obj.openStream());
```

Comment:
This change here means that the code will no longer follow cross-protocol redirects (eg http->https). By default, `HttpUrlConnection` _does_ follow redirects. However, it [_does not_ follow redirects across protocols](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=4620571).